File: benchmark/results/python-multilang/run-004/b-baseline/output/metaengine_demo/shipping/services/shipment_repository.py
"""ShipmentRepository for the shipping domain."""
from typing import Optional

from metaengine_demo.shipping.aggregates.shipment import Shipment


class ShipmentRepository:
    """Repository for persisting and querying Shipments."""

    def create(self, input: Optional[Shipment]) -> Shipment:
        """Persist a new Shipment."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Shipment]:
        """Find a Shipment by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Shipment]:
        """List Shipments up to limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a Shipment by id."""
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-invocation-inline/run-001/b-baseline/output/src/domain/shipping/services/ShipmentRepository.ts
import { Shipment } from "../aggregates/Shipment";

/** Repository abstraction for persisting and retrieving Shipments. */
export class ShipmentRepository {
  create(input: Partial<Shipment>): Shipment {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): Shipment | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Shipment[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}
